src/main/java/com/amazonaws/encryptionsdk/internal/Utils.java

```diff
@@ -311,4 +311,24 @@ public static String encodeBase64String(final byte[] data) {
         System.arraycopy(rawBytes, 0, paddedResult, length - rawBytes.length, rawBytes.length);
         return paddedResult;
     }
+
+    /**
+     * Returns true if the prefix of the given length for the input arrays are equal.
```

Comment:
Add a comment that this is not constant-time, or make this method constant time. Either is acceptable.

Reply:
Added comment